### ISSUE TITLE
[Feature Request]: Version 2.0.0: 'recordCrossOriginIframes' configuration is insecure.

### ISSUE THREAD

**Issue by BeichenPlus** [rrweb-io/rrweb#1190]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/rrweb-io/rrweb/issues) for a feature request that matches the one I want to file, without success.

### What package is this feature request for?

rrweb

### Problem Description

'window.parent.postMessage(message, '*');' is not safe.
If my page is referenced by an iframe from another site, then they can get my page content by including 'rrweb'

### Proposed Solution

I think a config should be added to allow users to define secure uri.

### Alternatives Considered

--

**Comment by eoghanmurray** [comment]
Yes, I don't think there would be any other way of automatically determining the correct 3rd party domain/origin other than a config setting.
Have you attempted to add the config setting, or is the system too intimidating?
A PR would be very welcome :)

**Comment by Juice10** [comment]
Thanks for submitting this feature request @BeichenPlus, I think it totally makes sense. If anyone is open to creating a PR for this I'd be happy to accept & merge it.
